lib/systems/src/MainChargeSystem.cpp

```diff
@@ -0,0 +1,28 @@
+/* Imports */
+#include "MainChargeSystem.h"
+
+namespace calc {
```

Comment:
calc is a pretty generic name. I know I tend to be overly nitpicky about names, but especially when lots and lots of people will be reading and studying this code, it's extremely important for names to be descriptive and self-explanatory. What are you calculating, and how will this namespace be used?